src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11767,62 +11767,62 @@ private bool CanStartConsequenceExpression()
 
         internal ExpressionSyntax ParseConsequenceSyntax()
         {
-            Debug.Assert(this.CurrentToken.Kind is SyntaxKind.DotToken or SyntaxKind.OpenBracketToken);
             ExpressionSyntax expr = this.CurrentToken.Kind switch
             {
                 SyntaxKind.DotToken => _syntaxFactory.MemberBindingExpression(this.EatToken(), this.ParseSimpleName(NameOptions.InExpression)),
                 SyntaxKind.OpenBracketToken => _syntaxFactory.ElementBindingExpression(this.ParseBracketedArgumentList()),
                 _ => throw ExceptionUtilities.Unreachable(),
             };
 
-            while (true)
+            // https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/expressions.md#1288-null-conditional-member-access
```

Comment:
i would literally put the grammar production in here.  to make it clear what dependent-access means.  i.e:

```c#
// null_conditional_member_access
//    : primary_expression '?' '.' identifier type_argument_list? dependent_access*
//    ;
```

Reply:
this would be particularly useful, because you'd also be updating the comments accordingly for the assignment work you're doing now.